modules/imgproc/src/resize.cpp

```diff
@@ -1481,10 +1481,111 @@ typedef VResizeNoVec VResizeLanczos4Vec_32f;
 
 #endif
 
+#if CV_SIMD128_PERMUTE > 0
+
+/*
+    Deinterleave two vectors into the same sized type.
+
+    A1 B1 A2 B2 .... A4 B4
+    and return expanded int32x4 of the form:
+    A1 ... A4
+    B1 ... B4
+*/
+template<typename T, typename VT>
+void inline v_load_expand_deinterlace(const T *ptr, VT &out_even, VT &out_odd)
+{
+    v_load_deinterleave(ptr, out_even, out_odd);
+}
+
+/*
+    Similar to above, but we must expand to the larger type and deinterleave the
+    expanded values.
+
+    The quick expand requires a 16B permute to quickly deinterlace. This operation
+    may not port efficiently to other vector engines.
+*/
+template<>
+void inline v_load_expand_deinterlace(const short *ptr, v_int32x4 &out_even, v_int32x4 &out_odd)
+{
+    static const v_uint8x16 perm(0,1,4,5,8,9,12,13,2,3,6,7,10,11,14,15);
+    v_int16x8 in = v_reinterpret_as_s16(v_permute_1(perm, v_reinterpret_as_u8(v_load(ptr))));
```

Comment:
I think it make sense to guard with `CV_SIMD128_PERMUTE` this part only. It's possible to substitute permute&expand with
```
v_int32x4 v_src = v_reinterpret_as_s32(v_load(ptr));
out_even = (v_src << 16) >> 16;
out_odd = v_src >> 16;
```
BTW the substitution use the same quantity of vector operations so probably it could give similar performance gain. @pmur Could you compare performance of both implementations for VSX?

Reply:
Ah, neat trick. I think that should perform similarly.

Reply:
Ah, neat trick. That should perform nearly identical.